stan/math/rev/core/operator_addition.hpp

```diff
@@ -79,7 +48,16 @@ class add_vd_vari final : public op_vd_vari {
  * @return Variable result of adding two variables.
  */
 inline var operator+(const var& a, const var& b) {
-  return {new internal::add_vv_vari(a.vi_, b.vi_)};
+  return make_callback_vari(a.vi_->val_ + b.vi_->val_,
+                            [avi = a.vi_, bvi = b.vi_](const auto& vi) mutable {
+                              if (unlikely(std::isnan(vi.val_))) {
+                                avi->adj_ = NOT_A_NUMBER;
+                                bvi->adj_ = NOT_A_NUMBER;
+                              } else {
+                                avi->adj_ += vi.adj_;
+                                bvi->adj_ += vi.adj_;
```

Comment:
Would this be faster if we just saved pointers to the adjoints themselves? They can't move and we're not using the input `vals_`.

Reply:
Oh nevermind. Var is just pointer to vari which is value + adjoint.